linkedlist/src/main/java/ua/hillel/LinkedList.java

```diff
@@ -1,69 +1,163 @@
 package ua.hillel;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 /**
  * Implement doubly linked list that is based on {@link Node<T>} class (you should implement it as well).
  * This is a simplified version of {@link java.util.LinkedList}.
  */
 public class LinkedList<T> implements List<T> {
+    private static class Node<T> {
+        T element;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(T element) {
+            this.element = element;
+            this.next = null;
+            this.prev = null;
+        }
+    }
 
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private Node<T> getNode(int index) {
+        Node<T> currNode = head;
+        for (int i = 0; i < index; i++) {
+            currNode = currNode.next;
+        }
+        return currNode;
+    }
 
     @Override
     public void add(T element) {
-
+        Node<T> newElement = new Node<>(element);
+        if (head == null) {
+            head = tail = newElement;
+        } else {
+            tail.next = newElement;
+            tail = newElement;
+        }
+        size++;
     }
 
     @Override
     public void add(int index, T element) {
-
+        Objects.checkIndex(index, size + 1);
+        Node<T> newElement = new Node<>(element);
+        if (head == null || index == size) {
+            add(element);
+        } else if (index == 0) {
+            newElement.next = head;
+            head = newElement;
+        } else {
+            Node<T> prevIndex = getNode(index - 1);
+            newElement.next = prevIndex.next;
+            prevIndex.next = newElement;
+        }
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Objects.checkIndex(index, size);
+        return getNode(index).element;
```

Comment:
You're looping in `getNode()` with the `i  < index` condition, so you'll receive the previous element at position `index - 1`, but not the one you're looking for.